### PR TITLE
Implement fill the form for job seeker

### PR DESCRIPTION
**Pull request by alexkocharyan01**

### CODE REVIEW

File: src/components/AreaOfSpecialization/AreaOfSpecialization.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  })
+
+  const { handleSubmit } = methods
+
+  const onSubmit = (data: IJobSeekerProps) => {
+    const areaSpecialization = {
+      ...data,
+      isActive: true,
+      postedAt: '2023-05-02',
+      status: 'Pending',
```
Comment (ArthurO2002) [inline src/components/AreaOfSpecialization/AreaOfSpecialization.tsx:56]: Remove this, BBE will be updated accordingly

File: src/manager/jobSeeker/jobSeeker.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+const jobSeekerManager = {
+  axiosInstance,
+  jobSeeker(data: IJobSeekerProps) {
+    return axiosInstance.post<IJobSeekerProps, IAxiosResponse<IJobSeekerProps>>(
+      `users/v1/job-seeker/application`,
+      {
+        ...data,
+      }
+    )
+  },
```
Comment (ArthurO2002) [inline src/manager/jobSeeker/jobSeeker.ts:15]: Jobseeker is not manager rename to applications

Comment (ArthurO2002) [reply src/manager/jobSeeker/jobSeeker.ts:15]: const baseURL = '/users'

File: src/pages/fill-the-form-job/index.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        </Button>
+      </div>
+      <div className="mx-auto flex w-full max-w-[380px] flex-col items-center">
+        <Tab.Group
+          selectedIndex={selectedIndex}
+          onChange={setSelectedIndex}
+        >
+          <Tab.List className="hidden">
+            <Tab>Tab 1</Tab>
+            <Tab>Tab 2</Tab>
```
Comment (ArthurO2002) [inline src/pages/fill-the-form-job/index.tsx:38]: Translate

File: src/components/AreaOfSpecialization/AreaOfSpecialization.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    setJobSeekerReview(areaSpecialization)
```
Comment (ArthurO2002) [inline src/components/AreaOfSpecialization/AreaOfSpecialization.tsx:59]: What is this for?

File: src/pages/fill-the-form-job/index.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { LeftIcon } from '@components/Icons/LeftIcon'
+import { Translation } from '@constants/translations'
+import { Tab } from '@headlessui/react'
+import { Button } from '@uiComponents/Button'
+
+const FillTheForm = () => {
+  const [t] = useTranslation()
+
+  const [selectedIndex, setSelectedIndex] = useState<number>(0)
+  const [jobSeekerReview, setJobSeekerReview] = useState<null | IJobSeekerProps>(null)
```
Comment (ArthurO2002) [inline src/pages/fill-the-form-job/index.tsx:16]: Not sure why we need this.
Any data that need global usage should be kept in hook form

File: src/pages/fill-the-form-job/index.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            <Tab.Panel>
+              <div className="mb-5 flex justify-between text-xl">
+                <p>{t(Translation.PAGE_FILL_THE_FORM_JOB_ONE_TITLE)}</p>
+                <p className="text-primary">2/2</p>
+              </div>
+              <div>
+                <FilTheFormJobReview
+                  jobSeekerReview={jobSeekerReview}
+                  setSelectedIndex={setSelectedIndex}
+                />
```
Comment (ArthurO2002) [inline src/pages/fill-the-form-job/index.tsx:63]: Too complex I think

File: src/redux/slices/areaSpecialization/middleware.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+const areaSpecialization = (params: IJobSeekerProps) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setJobSeekerLoading(true))
+    await API.jobSeeker.jobSeeker(params)
+
+    dispatch(setJobSeekerSubmitSuccess(true))
+  } catch (error) {
+    dispatch(setError((error as IError).response?.data?.status.message))
+  } finally {
+    dispatch(setJobSeekerLoading(false))
```
Comment (ArthurO2002) [inline src/redux/slices/areaSpecialization/middleware.ts:19]: Namings are wrong, why this function name areaSpecialization but it's submitting jobSeeker form?

File: src/redux/slices/areaSpecialization/slice.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+import { getInitialState } from './initialState'
+import reducers from './reducers'
+
+const slice = createSlice({
+  name: 'users',
+  initialState: getInitialState(),
+  reducers,
+})
+
```
Comment (ArthurO2002) [inline src/redux/slices/areaSpecialization/slice.ts:11]: Why slice name is users but it's in areaSpecialization file, also I think we should rename this file

File: src/pages/fill-the-form-job/index.tsx (hunk 6 above)
Comment (alexkocharyan01) [reply src/pages/fill-the-form-job/index.tsx]: done

File: src/redux/slices/areaSpecialization/middleware.ts (hunk 7 above)
Comment (alexkocharyan01) [reply src/redux/slices/areaSpecialization/middleware.ts]: done

File: src/redux/slices/areaSpecialization/slice.ts (hunk 8 above)
Comment (alexkocharyan01) [reply src/redux/slices/areaSpecialization/slice.ts]: done

File: src/pages/fill-the-form-job/index.tsx (hunk 5 above)
Comment (alexkocharyan01) [reply src/pages/fill-the-form-job/index.tsx]: done

File: src/components/AreaOfSpecialization/AreaOfSpecialization.tsx (hunk 4 above)
Comment (alexkocharyan01) [reply src/components/AreaOfSpecialization/AreaOfSpecialization.tsx]: i remove it

File: src/pages/fill-the-form-job/index.tsx (hunk 3 above)
Comment (alexkocharyan01) [reply src/pages/fill-the-form-job/index.tsx]: done

File: src/manager/jobSeeker/jobSeeker.ts (hunk 2 above)
Comment (alexkocharyan01) [reply src/manager/jobSeeker/jobSeeker.ts]: done

File: src/components/AreaOfSpecialization/AreaOfSpecialization.tsx (hunk 1 above)
Comment (alexkocharyan01) [reply src/components/AreaOfSpecialization/AreaOfSpecialization.tsx]: in this case we need

File: src/components/FilTheFormJobReview/FilTheFormJobData.tsx [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+export const FilTheFormJobReview = ({ setSelectedIndex }: IFilTheFormJobReviewProps) => {
+  const [t] = useTranslation()
+
+  const { getValues } = useFormContext()
+
+  const values = getValues()
+  const changeJobDetails = () => {
+    setSelectedIndex((prev) => prev - 1)
+  }
```
Comment (ArthurO2002) [inline src/components/FilTheFormJobReview/FilTheFormJobData.tsx:19]: goBack or something else

File: src/pages/fill-the-form-job/index.tsx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    dispatch(applicationsMiddleware.jobSeeker(areaSpecialization))
+  }
+
+  useEffect(() => {
+    if (isJobSeekerSubmitSuccess) {
+      dispatch(applicationsMiddleware.resetJobSeekerSubmitSuccess())
+      router.push('/')
+      reset(defaultValues)
```
Comment (ArthurO2002) [inline src/pages/fill-the-form-job/index.tsx:69]: Go to applications list page

**Pull request merged**